include/comm/Connection.h

```diff
@@ -68,6 +74,14 @@ class Connection
     uint32_t _max_buffer_size{};
 
     ConnMetrics* _metrics{nullptr};
+
+    std::string _source{};
+    std::string _encryption{};
+    short _source_family;  // from socket(2)
```

Comment:
Why duplicate these values here?  Can't we just get them from the socket?

Reply:
the values aren't duplicate here, this is where they are rendered into a uniform format. So my thinking is that if we were to make a unix socket connection at some point, it would be behind this abstraction, and this would be the point it would be unified for the metrics user of this data.

Reply:
I'm not sure I follow. It looks to me like these members are simply caching values returned by the socket. My question is why cache these values inside the base class?  Wouldn't a virtual interface give you the same abstraction with a smaller footprint per connection and potentially fewer string copies?  Eg.:
```cpp
class Connection
{
   public:
    // ...

    virtual std::string get_source() const = 0;
    virtual short get_source_family() const = 0;
    virtual std::string get_encryption() const = 0;
};


class TCPConnection : public Connection
{
   public:
    // ...

    std::string get_source() const override { return _tcp_socket->print_source(); }
    short get_source_family() const override { return _tcp_socket->source_family(); }
    std::string get_encryption() const { return "none"; }
};


class TLSConnection : public Connection { /* as above */ };


// possibly in the future
class UNIXConnection : public Connection
{
   public:
    // ...

    std::string get_source() const override { /*...*/ }
    short get_source_family() const override { /*...*/ }
    std::string get_encryption() const { /*...*/ }
};
```

Reply:
modified to be virtual interface